utils.js

```diff
@@ -0,0 +1,66 @@
+import globals from './globals';
+
+export const isObject = value => value !== null && typeof value === 'object';
+export const supportsAbortController = typeof globals.AbortController === 'function';
+export const supportsStreams = typeof globals.ReadableStream === 'function';
+
+export const mergeHeaders = (...sources) => {
+	const result = {};
+
+	for (const source of sources) {
+		if (!isObject(source)) {
+			throw new TypeError('The `headers` argument must be an object');
+		}
+
+		const headers = new globals.Headers(source);
+
+		for (const [key, value] of headers) {
+			// Headers constructor changes the value to a string
+			if (value === undefined || value === 'undefined') {
+				Reflect.deleteProperty(result, key);
+			} else {
+				Reflect.set(result, key, value);
+			}
+		}
+	}
+
+	return new globals.Headers(result);
+};
+
+export const deepMerge = (...sources) => {
+	let returnValue = {};
+
+	for (const source of sources) {
+		if (Array.isArray(source)) {
+			if (!(Array.isArray(returnValue))) {
+				returnValue = [];
+			}
+
+			returnValue = [...returnValue, ...source];
+		} else if (isObject(source)) {
+			for (let [key, value] of Object.entries(source)) {
+				if (isObject(value) && Reflect.has(returnValue, key)) {
+					if (key === 'headers') {
```

Comment:
It shouldn't be done like this. You check **every** object for `headers`. Something like this would be much better:

```js
let headers = {};

for (const source of sources) {
	if (Array.isArray(source)) {
		...
	} else if (isObject(source)) {
		...
	}

	if (source.headers) {
		headers = mergeHeaders(headers, source.headers);
	}
}

returnValue.headers = headers;
```

Reply:
You're right, I'll update the PR.